### ISSUE TITLE
Add option to not fail if wanted is met

### ISSUE THREAD

**Issue by mdarefull** [jens-duttke/check-outdated#7]
I think it would be very useful if we could pass an argument to the command to not fail if all the wanted versions are met.

By ignoring a package we won't have the opportunity to ensure the wanted versions are met. This would allow us to leverage the ^ and ~ version range specifier and fail our build if our versions don't meet them.

**Comment by jens-duttke** [comment]
@mdarefull I'm not sure if I understand you correctly. Could you please provide an example?

**Comment by mdarefull** [comment]
Of course, 

I use the library as part of my PR build to enforce developers to always update our dependency libraries. Under some circumstances we might not be able to update to the latest version of a dependency, for those scenarios we use the version range specifier to determine what versions we can afford to update.
Ex.: "1.2.3" would mean we cannot update this package while "^1.2.3" means we can update to 1.3.0 is available.

If I run check-outdated as part of my CI, it fails if package X meets the wanted version but not the latest. 
My particular case is angular with typescript. The angular builder supports a very restrictive range of typescript versions, usually meaning that 90% of your time, you will have a wanted ts version, but not the latest one.

Currently, we tell check-outdated to ignore typescript, but then we are missing on those scenarios where an acceptable version is available while also increases the maintenance cost because now we have to be excluding and including back packages based on version support.

In summary:
If I run check-outdated:
![image](https://user-images.githubusercontent.com/22510674/112011446-0a5f9400-8b07-11eb-8d82-a01dd3dd3749.png)
I think it would be very valuable to have an option allowing the command to fail driven by the "range specifiers" on the package versions: 
* The command fails only if a wanted version is not met.
* It won't fail for `"typescript": "~4.1.5"` if `4.2.0` is available, because I'm specifying that I only want `4.1.*`.
* It won't ever fail for `"typescript": "4.1.5"`. This is similar to ignoring the package, but I only manage this in the package.json file and not also on my CI pipelines.

**Comment by jens-duttke** [comment]
Thanks for the clarification!

I think this should be the default behaviour and an option `--strict` (or something similar), should be the current behaviour.
I'm always using fixed version numbers, and no ranges, since not all packages are using semantic versioning ([also not TypeScript](https://github.com/microsoft/TypeScript/issues/14116)), but at the end it's up to the user to use ranges, if they want - so `check-outdated` should support that correctly.

I'll implement a solution for that within the next days.

**Comment by jens-duttke** [comment]
After rethinking the requirement and playing a bit around, I decided not to make this the default behaviour.

So I've added a new argument `--prefer-wanted`, which ensures that your installed version matches the latest "Wanted" version.
